core/src/main/java/org/apache/carbondata/core/metadata/schema/table/DataMapSchema.java

```diff
@@ -67,6 +92,83 @@ public void setRelationIdentifier(RelationIdentifier relationIdentifier) {
 
   public void setChildSchema(TableSchema childSchema) {
     this.childSchema = childSchema;
+    List<ColumnSchema> listOfColumns = this.childSchema.getListOfColumns();
+    fillNonAggFunctionColumns(listOfColumns);
+    fillAggFunctionColumns(listOfColumns);
+    fillParentNameToAggregationMapping(listOfColumns);
```

Comment:
I feel all the above 3 functions doing the almost same job, why don't you combine all of them.